source/SharpGL/Core/SharpGL.WPF/OpenGLControl.xaml.cs

```diff
@@ -31,21 +23,72 @@ public partial class OpenGLControl : UserControl
         public OpenGLControl()
         {
             InitializeComponent();
-            SizeChanged += new SizeChangedEventHandler(OpenGLControl_SizeChanged);
+
+            Unloaded += OpenGLControl_Unloaded;
+            Loaded += OpenGLControl_Loaded;
+        }
+
+        /// <summary>
+        /// Handles the Loaded event of the OpenGLControl control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="System.Windows.RoutedEventArgs"/> Instance containing the event data.</param>
+        private void OpenGLControl_Loaded(object sender, RoutedEventArgs routedEventArgs)
+        {
+            SizeChanged += OpenGLControl_SizeChanged;
+
+            if (timer != null)
+            {
+                timer.Tick -= timer_Tick;
+                timer.Tick += timer_Tick; // to avoid twice call, ensure timer_Tick event handler is registered only once, because it is also registred on apply template
+            }
+
+            UpdateOpenGLControl((int) RenderSize.Width, (int) RenderSize.Height);
+
+            //  DispatcherTimer setup
+            timer = new DispatcherTimer();

```

Comment:
If we create the timer in the constructor, we never need to check it for null (which simplifies line 63). We should set the `Interval` property in `OnTemplateApplied`. Then in `Loaded` we `Stop` then unregister, and in loadad we register and `start`. This makes `Loaded` and `UnLoaded` a bit clearer, as the only 'activate' an 'deactive' the timer, rather than configure it as well.